addons/pos_self_order/models/pos_config.py

```diff
@@ -184,10 +270,41 @@ def _get_self_order_data(self) -> Dict:
             "show_prices_with_tax_included": self.iface_tax_included == "total",
             "custom_links": self._get_self_order_custom_links(),
             "products": self._get_available_products()._get_self_order_data(self),
-            "pos_category": self.env['pos.category'].search_read(fields=["name", "sequence"], order="sequence"),
+            "combos": self._get_combos_data(),
+            "pos_category": self.env['pos.category'].search_read(fields=["name", "sequence", "has_image"], order="sequence"),
             "has_active_session": self.has_active_session,
         }
 
+    def _get_combos_data(self):
+        self.ensure_one()
+        combos = self.env["pos.combo"].search([])
+
+        return[{
+            'id': combo.id,
+            'name': combo.name,
+            'combo_line_ids': combo.combo_line_ids.read(['product_id', 'price', 'lst_price', 'combo_id'])
+        } for combo in combos]
+
+    def _get_self_order_kiosk_data(self):
```

Comment:
not sure it should have an underscore at the start of the function as it is used outside the class

Reply:
I think no underscore is only when the frontend can call it